pkg/scheduler/factory/factory.go

```diff
@@ -189,6 +189,8 @@ type configFactory struct {
 	storageClassLister storagelistersv1.StorageClassLister
 	// a means to list all CSINodes
 	csiNodeLister storagelistersv1beta1.CSINodeLister
+	// a means to list all CSIDrivers
+	csiDriverLister storagelistersv1beta1.CSIDriverLister
```

Comment:
You should consider making the new changes a plugin for the new scheduling framework. We have already placed a lot of code related to CSI and volume binding in the core parts of the scheduler. Instead of adding more, we should consider removing the existing ones and converting them into plugins. New additional logic should also be added as new plugins.

cc/ @ahg-g

Reply:
Cool, I'll look into that. Do you have in mind any existing plugins that I could use as an example?

Reply:
you will be the first :)

There are few examples here: https://github.com/kubernetes/kubernetes/tree/master/pkg/scheduler/framework/plugins/examples

Reply:
Do we know how these plugins will be shipped? Are they part of default k8s build or they are build and shipped separately?

Reply:
We will have a set of default plugins which are built inside K8s/K8s repo, shipped with the code, and enabled by default. We will also have a second tier of plugin which are shipped with the code, but are not enabled by default. Users may also build their own plugins in their private repo.
This particular feature should probably be one of those plugins and is built, shipped, and enabled by default.